### DISCUSSION TITLE
Display longer hash

### DISCUSSION THREAD

**Discussion by KevinWuWon** [arxanas/git-branchless#279] [Ideas]
A few times, the 8-character hash has been too short and it has given me a collision. It's quite annoying when this happens because it just gives me an error message like:

```
Git error Ambiguous: ambiguous SHA1 prefix - multiple matches for prefix: 1e0397ec4cc2148c6587e30bc85baae8b3f26002 1e0397ecb633e9115514b7c256560ddc27594289
```

and I still don't know which one I want.

My two feature requests are:

1. in this error message, print also the commit title so I can tell them apart
2. give an option to increase the default hash prefix length to display in `git sl`

**Comment by epage** [comment]
Maybe I misunderstood but can't git2 abbreviate for you, extending lengths when running into ambiguities.  Is there a reason you don't use this in git-branchess?

From `git-stack`:
```rust
let abbrev_id = self
    .repo
    .raw()
    .find_object(node.commit.id, None)
    .unwrap()
    .short_id()
    .unwrap();
```

**Reply by arxanas** [reply]
I'm pretty sure using the `git2` function works, I just never got around to implementing/fixing it.

**Reply by arxanas** [reply]
It should be a relatively straightforward refactoring if you want to try @KevinWuWon. You just need to pass in `repo` to the constructor here and use it in the `describe` function:  https://github.com/arxanas/git-branchless/blob/master/src/core/node_descriptors.rs#L85

**Comment by arxanas** [comment]
Thanks to @yujong-lee's change in #306, this should be fixed on the latest development build :)
